core/server/common/src/main/java/alluxio/master/StateLockManager.java

```diff
@@ -169,6 +178,8 @@ public LockResource lockExclusive(StateLockOptions lockOptions)
       }
     }
     if (lockAcquired) { // Lock was acquired within grace-cycle.
+      LOG.info("Thread-{} acquired the lock within grace-cycle.",
+          ThreadUtils.getCurrentThreadIdentifier());
       activateInterruptCycle();
     } else { // Lock couldn't be acquired by grace-cycle.
       if (graceMode == StateLockOptions.GraceMode.TIMEOUT) {
```

Comment:
Hrmmm, is this right? I thought TIMEOUT mode would not force the lock by interrupting threads?

Reply:
no we have to interrupt during the lock. not just when forcing the lock.

Reply:
If the lock is acquired by the grace-cycle, we still need to run the interrupt-cycle for maintaining the heart-beats.

TIMEOUT and FORCED make sense when the lock was not grabbed by the grace-cycle. In this context FORCED would mean to activate the interrupt-cycle before taking the lock.